docs/content/docs/reference/react/hooks.mdx

```diff
@@ -0,0 +1,393 @@
+---
+title: 'Hooks'
+---
+
+## Authentication Hooks
+
+### Email + password registration
+
+```js
+const { signUp, isLoading, isSuccess, needsVerification, isError, error } =
+  useEmailPasswordSignUp(email?: string, password?: string, options?: SignUpOptions )
+```
+
+- `signUp: (email?: string, password?: string) => void` triggers the user registration. The email/password arguments will take precedence over the possible state values used when creating the hook.
+- `isLoading: boolean` returns `true` when the action is executing, `false` when the action result is available.
+- `needsVerification: boolean` returns `true` if the registration has been accepted, but a verificaiton email has been sent and is awaiting.
+- `isSuccess: boolean` returns `true` if the registration suceeded. Returns `false` if the new email needs to be verified first, or if an error occurred.
+- `isError: boolean` returns `true` if an error occurred.
+- `error?: {status: number, error: string, message: string}` provides details about the error.
+- `SignUpOption`:
+  - `locale?: string` locale of the user, in two digits, for instance `en`.
+  - `allowedRoles?: string[]`: allowed roles of the user. Must be a subset of the default allowed roles defined in Hasura Auth.
+  - `defaultRole?: string`: default role of the user. Must be part of the default allowed roles defined in Hasura Auth.
+  - `displayName?: string`
+  - `metadata?: Record<string, unknown>`: Custom additional user information stored in the `metadata` column.
+  - `redirectTo?: string`: redirection path in the client application that will be used in the link in the verification email. For instance, if you want to redirect to `https://myapp.com/success`, the `redirectTo` value is `'/success'`.
```

Comment:
Is this accurate? I thought you would have to specify the full URL for `redirectTo`.

Reply:
I don't think so.

In Hasura Auth we grab the `redirectTo` from the query parameter [here](https://github.com/nhost/hasura-auth/blob/main/src/routes/verify/verify.ts#L12) and then send a redirect [here](https://github.com/nhost/hasura-auth/blob/main/src/routes/verify/verify.ts#L89-L91).

Reply:
the xstate-based client prepends any route starting with a `/` with the client URL, which is by default `window.location.origin`. the full path is always send to hasura-auth